zip-proportional-output-fee-mechanism-pofm.rst

```diff
@@ -0,0 +1,262 @@
+
+
+Requirements
+============
+
+* The conventional fee formula should not favour or discriminate against any
+  of the Orchard, Sapling, or transparent protocols.
+* The fee for a transaction should scale linearly with the number of inputs
+  and/or outputs.
+* Users should not be penalised for sending transactions constructed
+  with padding of inputs and outputs to reduce information leakage.
+  (The default policy employed by zcashd and the mobile SDKs pads to
+  two inputs and outputs for each shielded pool used by the transaction).
+* Users should be able to spend a small number of UTXOs or notes with value
+  below the marginal fee per input.
+
+
+Specification
+=============
+
+This specification defines three parameters that are used to calculate the
+conventional fee:
+
+=================== ============================
+Parameter           Units
+=================== ============================
+`marginal_fee`      zatoshis per input or output
+`grace_window_size` inputs or outputs
+=================== ============================
+
+Wallets implementing this specification SHOULD use a conventional fee
+calculated in zatoshis per the following formula::
+
+    logical_actions = max(transparent_inputs, transparent_outputs) +
+                      2*sprout_joinsplits +
+                      max(sapling_inputs, sapling_outputs) +
+                      orchard_actions
+
+    conventional_fee = marginal_fee * max(grace_actions, logical_actions)
+
+The parameters are set to the following values:
+* `marginal_fee = 5000`;
+* `grace_window_size = 2`.
+
+It is not a consensus requirement that fees follow this formula; however,
+wallets SHOULD create transactions that pay this fee, in order to reduce
+information leakage, unless overridden by the user.
+
+Transaction relaying
+--------------------
+
+zcashd, zebrad, and potentially other node implementations, implement
+fee-based restrictions on relaying of mempool transactions. Nodes that
+normally relay transactions are expected to do so for transactions that pay
+at least the conventional fee as specified in this ZIP, unless there are
+other reasons not to do so for robustness or denial-of-service mitigation.
+
+Mempool size limiting
+---------------------
+
+zcashd and zebrad limit the size of the mempool as described in [#zip-0401]_.
+This specifies a *low\_fee\_penalty* that is added to the "eviction weight"
+if the transaction pays a fee less than the conventional transaction fee.
+This threshold is modified to use the new conventional fee formula.
+
+Block production
+----------------
+
+Miners, mining pools, and other block producers, select transactions for
+inclusion in blocks using a variety of criteria. Where the criteria
+previously used the conventional transaction fee defined in ZIP 313 to
+decide on transaction inclusion, it is expected to instead use the formula
+specified in this ZIP.
+
+Open Issues
+-----------
+
+> TODO: Remove this section once a decision is made.
```

Comment:
This section will need to become a Rationale section briefly saying why we chose these parameters and this formula. When describing the earlier proposed parameters, we will need to take into account that they are not directly comparable with the specified `marginal_fee`, because they were multiplied by `inputs + outputs` instead of logical actions.